src/uri/port.rs

```diff
@@ -51,14 +51,32 @@ impl<'a> Port<'a> {
     }
 }
 
+impl<'a> fmt::Display for Port<'a> {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(fmt, "{}", self.bytes)
+    }
+}
+
 impl<'a> PartialEq for Port<'a> {
     fn eq(&self, other: &Self) -> bool {
         self.port == other.port
     }
 }
 
-impl<'a> fmt::Display for Port<'a> {
-    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
-        write!(fmt, "{}", self.bytes)
+impl<'a, T: From<u16> + PartialEq> PartialEq<T> for Port<'a> {
```

Comment:
I would rather avoid a blanket impl for this partialeq. I believe you did this to avoid having to add impls for each u* type. Perhaps a macro would work instead?

The problem is that adding this blanket impl would prevent us from adding a different blanket impl. Perhaps we would like to `impl PartialEq<T>` for `T: AsRef<Port>` or something.